src/dbwrapper.cpp

```diff
@@ -71,20 +71,40 @@ class CBitcoinLevelDBLogger : public leveldb::Logger {
     }
 };
 
+static void SetMaxOpenFiles(leveldb::Options *options) {
+    // On most platforms the default setting of max_open_files (which is 1000)
+    // is optimal. On Windows using a large file count is OK because the handles
+    // do not interfere with select() loops. On 64-bit Unix hosts this value is
+    // also OK, because up to that amount LevelDB will use an mmap
+    // implementation that does not use extra file descriptors (the fds are
+    // closed after being mmaped).
+    //
+    // Increasing the value beyond the default is dangerous because LevelDB will
+    // fall back to a non-mmap implementation when the file count is too large.
+    // On 32-bit Unix host we should decrease the value because the handles use
+    // up real fds, and we want to avoid fd exhaustion issues.
+    //
+    // See PR #12495 for further discussion.
+#ifndef WIN32
+    if (sizeof(void*) < 8)
+        options->max_open_files = 64;
+#endif
+}
+
 static leveldb::Options GetOptions(size_t nCacheSize)
 {
     leveldb::Options options;
     options.block_cache = leveldb::NewLRUCache(nCacheSize / 2);
     options.write_buffer_size = nCacheSize / 4; // up to two write buffers may be held in memory simultaneously
     options.filter_policy = leveldb::NewBloomFilterPolicy(10);
     options.compression = leveldb::kNoCompression;
-    options.max_open_files = 64;
```

Comment:
max_open_files is not set explicitly before calling SetMaxOpenFiles. I think you mean to set this to 1000 instead of removing the line?

Reply:
The constructor for `leveldb::Options` sets it to 1000.

Reply:
I didn't scroll far another down :) - you're right.